modules/sage-aws-k8s-node-autoscaler/main.tf

```diff
@@ -102,6 +102,35 @@ module "ocean-aws-k8s" {
   is_aggressive_scale_down_enabled = true
   max_scale_down_percentage        = 33
   tags                             = var.tags
+  # TODO: Fix this it does not seem to work
+  # `desired_capacity` does not seem to force the number of nodes to increase. Look
+  # through the documentation to determine how we might manually scale up the number
+  # of nodes if we wanted to.
+  desired_capacity = var.desired_capacity
+
+
+  filters = {
+    exclude_metal = true
+    hypervisor    = ["nitro"]

```

Comment:
I ran into issues when I deployed airflow where the number of pods maxed out the new of private IP addresses that I could assign to the node. This should default to instances that allow trunk based ENIs, along with the `ENABLE_PREFIX_DELEGATION` in the CNI add on this should let more pods be assignable to each node.

After setting this up I could see that from https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/v1.5.0/pkg/aws/vpc/limits.go - This m5.large was the same instance size I originally ran into the problem with (`m6a.large`). So it confirms that this does work.
```
	"m5.large": {
		Interface:               3,
		IPv4PerInterface:        10,
		IsTrunkingCompatible:    true,
		BranchInterface:         9,
		DefaultNetworkCardIndex: 0,
		NetworkCards: []NetworkCard{
			{
				MaximumNetworkInterfaces: 3,
				NetworkCardIndex:         0,
			},
		},
		Hypervisor:  "nitro",
		IsBareMetal: false,
	},
```